Remotion/Web/Core/WebStringConverter.cs

```diff
@@ -0,0 +1,148 @@
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.ComponentModel;
+using System.ComponentModel.Design.Serialization;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Remotion.Utilities;
+
+namespace Remotion.Web
+{
+  /// <summary>
+  /// Provides <see cref="WebString"/> conversion from and to <see cref="String"/> and <see cref="InstanceDescriptor"/> instances.
+  /// </summary>
+  public class WebStringConverter : TypeConverter
+  {
+    private const string c_textPrefix = "(text)";
+    private const string c_htmlPrefix = "(html)";
+
+    private static readonly Lazy<MethodInfo> s_createFromTextMethodInfo = new(
+        () =>
+        {
+          var methodInfo = typeof (WebString).GetMethod (nameof (WebString.CreateFromText), BindingFlags.Static | BindingFlags.Public)!;
+          Assertion.DebugIsNotNull (
+              methodInfo.GetParameters().SingleOrDefault (p => p.ParameterType == typeof (string)),
+              "The parameters of '{0}' have changed, please adapt '{1}' accordingly.",
+              nameof (WebString.CreateFromText),
+              nameof (WebStringConverter));
+          return methodInfo;
+        },
+        LazyThreadSafetyMode.ExecutionAndPublication);
+
+    private static readonly Lazy<MethodInfo> s_createFromHtmlMethodInfo = new(
+        () =>
+        {
+          var methodInfo = typeof (WebString).GetMethod (nameof (WebString.CreateFromHtml), BindingFlags.Static | BindingFlags.Public)!;
+          Assertion.DebugIsNotNull (
+              methodInfo.GetParameters().SingleOrDefault (p => p.ParameterType == typeof (string)),
+              "The parameters of '{0}' have changed, please adapt '{1}' accordingly.",
+              nameof (WebString.CreateFromHtml),
+              nameof (WebStringConverter));
+          return methodInfo;
+        },
+        LazyThreadSafetyMode.ExecutionAndPublication);
+
+    /// <inheritdoc />
+    public override bool CanConvertFrom (ITypeDescriptorContext context, Type sourceType)
+    {
+      return sourceType == typeof (string)
+             || base.CanConvertFrom (context, sourceType);
+    }
+
+    /// <inheritdoc />
+    public override bool CanConvertTo (ITypeDescriptorContext context, Type destinationType)
+    {
+      return destinationType == typeof (string)
+             || destinationType == typeof (InstanceDescriptor)
+             || base.CanConvertTo (context, destinationType);
+    }
+
+    /// <inheritdoc />
+    public override object? ConvertFrom (ITypeDescriptorContext context, CultureInfo culture, object? value)
+    {
+      if (value == null)
+        return null;
+
+      if (value is string @string)
+        return ConvertFromStringInner (@string);
```

Comment:
As `TypeConverter` defines `ConvertFromString` I decided against the `new` keyword and the name conflict. But I'll change it.

Reply:
ah, okay, didn't know that. yeah, `new` is perhaps better? Or some other name? Perhaps `CovnertFromStringToWebString`?

I don't like `Inner` since it's just noise. I sometimes use `...Implementation` if there's a conflict I need to solve and lack creativity. `inner` is more for decorator scenarios.